fivox/spikeLoader.cpp

```diff
@@ -60,62 +59,40 @@ class SpikeLoader::Impl
         }
 
         const std::string& spikePath = params.getSpikes();
-        _loadSpikes( spikePath.empty() ? params.getConfig().getSpikeSource()
-                                       : URI( spikePath ));
-
+        _report.reset(
+            new brain::SpikeReportReader( spikePath.empty()
+                                          ? params.getConfig().getSpikeSource()
+                                          : URI( spikePath ),
+                                          gids ));
+        _spikesEnd = _report->getEndTime();
+        // We need _hasEnded to return true  is file based reports because
+        // otherwise getFrameRange doesn't work as expected before the end
+        // of the report is reached. The only way to detect this with the new
+        // API is by checking if the endTime is already different from 0 at
+        // construction, because for streams it will always be 0. This is
+        // relying on an implementation detail of the spikeReport, so it's
+        // worth considering to include this detail as part of the API or add
+        // the isStream function again;
```

Comment:
This indicates that it's more difficult than it seems to have a common API that allows random access when the underlying IO can be both a file or a stream. You all opposed the isStream function, but to me it actually made things clearer.
Note that this is using SpikeReportReader and not SpikeReport, which has been modified only to remove functions (getStartTime and isStream). It has to be SpikeReportReader because in the stream case you want to keep the spikes collected for future random access.

I don't think start time and end time were 0 before.
If you want to see funny things I can point you to other places I don't fully understand and didn't dare to change.